application-diagram-api/src/main/java/com/xwiki/diagram/internal/DiagramApplicationListener.java

```diff
@@ -60,26 +70,59 @@ public class DiagramApplicationListener extends AbstractEventListener
     @Inject
     private WikiDescriptorManager wikiDescriptorManager;
 
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    private ComponentManager componentManager;
+
     /**
      * Default constructor.
      */
     public DiagramApplicationListener()
     {
-        super(ROLE_HINT, new ExtensionUpgradedEvent("com.xwiki.diagram:application-diagram"));
+        super(ROLE_HINT,
+            Arrays.<Event>asList(new ExtensionUpgradedEvent(DIAGRAM_APPPLICATION_ID), new ExtensionInstalledEvent()));
```

Comment:
Why do you use a filter for the ``ExtensionUpgradedEvent`` and not for ``ExtensionInstalledEvent`` (especially since you do the filtering yourself below with the same extension id). Deserves at least a comment.

Reply:
`ExtensionInstalledEvent` does not have a constructor that matches the specified extension id on all namespaces, so I will add a comment to explain it
see https://github.com/xwiki/xwiki-commons/blob/master/xwiki-commons-core/xwiki-commons-extension/xwiki-commons-extension-api/src/main/java/org/xwiki/extension/event/ExtensionInstalledEvent.java and https://github.com/xwiki/xwiki-commons/blob/master/xwiki-commons-core/xwiki-commons-extension/xwiki-commons-extension-api/src/main/java/org/xwiki/extension/event/ExtensionUpgradedEvent.java#L84